File: lib/java/src/main/java/io/github/brenbar/uapi/internal/GenerateRandomNumber.java
package io.github.brenbar.uapi.internal;

import io.github.brenbar.uapi.RandomGenerator;

public class GenerateRandomNumber {
    static Object generateRandomNumber(Object blueprintValue, boolean useBlueprintValue,
            RandomGenerator randomGenerator) {
        if (useBlueprintValue) {
            return blueprintValue;
        } else {
            return randomGenerator.nextDouble();
        }
    }
}

File: lib/go/internal/types/GenerateRandomNumber.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

// GenerateRandomNumber returns a pseudo-random floating-point number, optionally reusing a blueprint value.
func GenerateRandomNumber(blueprintValue any, useBlueprintValue bool, ctx *GenerateContext) any {
	if useBlueprintValue {
		return blueprintValue
	}
	if ctx == nil || ctx.RandomGenerator == nil {
		return 0.0
	}
	return ctx.RandomGenerator.NextDouble()
}
